### ISSUE TITLE
Support service filter on nomad service discovery

### ISSUE THREAD

**Issue by quentin9696** [prometheus/prometheus#11465]
### Proposal

Hi, 

I'm trying to use the nomad service discovery at home and face to a missing feature. In the `nomad_sd_config`, we can't filter by service. 

I saw in the API we can select only one service by using GET /v1/service/:service_name (https://www.nomadproject.io/api-docs/services#read-service). 

A possible implementation can be `services: list of string` as it's already done in the `consul_sd_config`

**Comment by roidelapluie** [comment]
You should be able to filter the service with relabeling .

Le sam. 15 oct. 2022, 04:38, quentin9696 ***@***.***> a
écrit :

**Comment by quentin9696** [comment]
@roidelapluie yes, but I don't want to scrape everything. I just want my prometheus to scrape only few services. I already add a relabel to keep specific services

**Comment by roidelapluie** [comment]
There is a difference between relabel_configs and metric_relabel_configs . Can you share your configuration? (Without secrets)

**Comment by quentin9696** [comment]
Hi, 

Here is my config, 

``` yaml
---
global:
  scrape_interval:     5s
  evaluation_interval: 5s

scrape_configs:
- job_name: openweathermap
  scrape_interval: 10m

  nomad_sd_configs:
  - server: "http://127.0.0.1:4646"

  metric_relabel_configs:
   - source_labels: [ __meta_nomad_service ]
     regex: 'openweather'
     action: keep
```

And here is my targets: 
The target with port `21716` is my prometheus and the other is the one I want to scrape.

![image](https://user-images.githubusercontent.com/14543538/196012396-121b95b8-4a71-4baf-ae50-a17156530353.png)

**Comment by quentin9696** [comment]
Hi, 

I come back with this feature-request again. 

I have added more service in my nomad cluster and facing now an issue: I want to relabel only some specifics services, and that become to be a pain de manage. 

I also check if it's possible to create my own pull request but I never touch go and don't see where the API calls are done. I think you're using the nomad SDK, but not sure where

Quentin

**Comment by beorn7** [comment]
Hello from the bug scrub!

This looks to us as if `relabel_configs` is still not used in your config. (It _does_ have the capability of not scraping certain targets.) Maybe you are fishing for something else, though, which might or might not be addressed by #11775 (which does have an open PR to add a feature to Nomad SD). Let's close this and continue any discussion on #11775.
